tests/components/evohome/test_init.py

```diff
@@ -28,3 +36,72 @@ async def test_entities(
     await setup_evohome(hass, config, install=install)
 
     assert hass.states.async_all() == snapshot
+
+
+@pytest.mark.parametrize(
+    "status", [*sorted([*_ERR_MSG_LOOKUP_AUTH, HTTPStatus.BAD_GATEWAY]), None]
+)
+async def test_authentication_failure_v2(
+    hass: HomeAssistant,
+    config: dict[str, str],
+    status: HTTPStatus,
+    caplog: pytest.LogCaptureFixture,
+) -> None:
+    """Test failure to setup an evohome-compatible system."""
+
+    with patch("evohomeasync2.broker.Broker.get") as mock_fcn:
+        mock_fcn.side_effect = exc.AuthenticationFailed("", status=status)
+
+        with caplog.at_level(logging.WARNING):
+            result = await async_setup_component(hass, DOMAIN, {DOMAIN: config})
+
+    assert result is False
+
+    # ignore any logging from the client library
+    log = [r.message for r in caplog.records if r.name.startswith("homeassistant.")]
+
+    assert log[0].startswith("Failed to authenticate")  # this entry is from evohome
+    assert log[1].startswith("Setup failed for")
+    assert len(log) == 2
+
+
+@pytest.mark.parametrize(
+    "status", [*sorted([*_ERR_MSG_LOOKUP_BASE, HTTPStatus.BAD_GATEWAY]), None]
+)
+async def test_client_request_failure_v2(
+    hass: HomeAssistant,
+    config: dict[str, str],
+    status: HTTPStatus,
+    caplog: pytest.LogCaptureFixture,
+) -> None:
+    """Test failure to setup an evohome-compatible system."""
+
+    with patch("evohomeasync2.broker.Broker.get") as mock_fcn:
+        mock_fcn.side_effect = exc.RequestFailed("", status=status)
+
+        with caplog.at_level(logging.WARNING):
+            result = await async_setup_component(hass, DOMAIN, {DOMAIN: config})
+
+    assert result is False
+
+    # ignore any logging from the client library
+    log = [r.message for r in caplog.records if r.name.startswith("homeassistant.")]
+
+    if status in (
+        HTTPStatus.TOO_MANY_REQUESTS,
+        HTTPStatus.SERVICE_UNAVAILABLE,
+        None,
+    ):
+        assert log[1].startswith("Setup failed for")
+        assert len(log) == 2
+    else:
+        assert log[0].startswith("Error during setup of")
+        assert len(log) == 1
+
+    # these entries are from evohome
+    if status == HTTPStatus.TOO_MANY_REQUESTS:
+        assert "API rate limit" in log[0]
+    elif status == HTTPStatus.SERVICE_UNAVAILABLE:
+        assert "currently unavailable" in log[0]
+    elif status is None:
+        assert "Unable to connect" in log[0]
```

Comment:
If statements should be avoided in tests. Please rewrite the test.

For example, you could add the expected log entries as additional parameterized test param